frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -264,6 +279,18 @@ export type CreateStorageSystemAction =
       type: 'securityAndNetwork/setKms';
       payload: WizardState['securityAndNetwork']['kms'];
     }
+  | {
+      type: 'securityAndNetwork/setVault';
+      payload: WizardState['securityAndNetwork']['kms']['vault'];
+    }
+  | {
+      type: 'securityAndNetwork/setIbmKms';
+      payload: WizardState['securityAndNetwork']['kms']['ibmKms'];
+    }
+  | {
+      type: 'securityAndNetwork/setKmsProvider';
+      payload: WizardState['securityAndNetwork']['kms']['kmsProvider'];
+    }
```

Comment:
Why not using `'securityAndNetwork/setKms'

Reply:
`securityAndNetwork/setKms` is used to updated entire KMS object (which contains state of all KMS providers, IBM/Vault etc, could be more in future). It is used for the case where we need to reset the entire KMS state (of all providers).

 `securityAndNetwork/setIbmKms` & `securityAndNetwork/setVault` are KMS provider specific. No need to pass entire KMS object, only pass the value of the particular provider we want to update.